crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -363,9 +365,11 @@ def prepare_nodes_for_crowbar_upgrade
         service_object = CrowbarService.new(Rails.logger)
         service_object.prepare_nodes_for_crowbar_upgrade
 
+        ::Crowbar::UpgradeStatus.new.end_step
         true
       rescue => e
         message = e.message
+        ::Crowbar::UpgradeStatus.new.end_step(false, message)
```

Comment:
The second parameter to `end_step` defaults to `{}`, so if `message` is a `String` then this looks like a bug.  It should not sometimes be a `Hash` and other times a `String`.